api/envoy/extensions/common/ratelimit/v3/ratelimit.proto

```diff
@@ -91,4 +91,39 @@ message RateLimitDescriptor {
 
   // Optional rate limit override to supply to the ratelimit service.
   RateLimitOverride limit = 2;
+
+  // Optional Token Bucket algorithm for local ratelimiting.
+  type.v3.TokenBucket token_bucket = 3;
+}
+
+message Entry {
+  // Descriptor key.
+  string key = 1 [(validate.rules).string = {min_len: 1}];
+
+  // Descriptor value.
+  string value = 2 [(validate.rules).string = {min_len: 1}];
+}
+
+message LocalRateLimitDescriptor {
+  // Descriptor entries.
+  RateLimitDescriptor descriptor = 1;
```

Comment:
In looking at this more I would just inline key and value in here or use RateLimitDescriptor.Entry